test/integration/page-objects/page.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+var extend = require('../helpers/extend').extend;
+var h = require('../helpers');
+var logger = require('../helpers/logger');
+
+var Page = function(project) {
+  this.project = project;
+};
+
+var clickHelpMenuChild = function(toggleSelector, menuSelector, itemIndex) {
+  var toggle = element(toggleSelector);
+  return toggle
+          .click()
+          .then(function() {
+            return element(menuSelector)
+                    .all(by.tagName('a'))
+                    .get(itemIndex)
+                    .click();
+          });
+};
+
+var clickSideMenuChild = function(mainMenuSelector, mainIndex, childMenuSelector, childIndex) {
+  var primaryMenuItem = element(mainMenuSelector.row(mainIndex));
+  return primaryMenuItem
+          .click()
+          .then(function() {
+            return primaryMenuItem
+                    .all(childMenuSelector)
+                    .get(childIndex)
+                    .click();
+          });
+};
+
+extend(Page.prototype, {
+  // individual pages should override the .getUrl function to return an appropriate path.
+  getUrl: function() {},
+  // FooPage.visit() is the preferred way to navigate to any page,
+  // which will work so long as it overrides .getUrl();
+  visit: function() {
+    logger.log('visiting url:',this.getUrl());
+    return h.goToPage(this.getUrl());
+  },
+  // TODO: decide if these should be kept for testing purposes, or removed.
+  // Page.topMenu & Page.sideMenu provide navigation via clicking the menu.
+  // HOWEVER, prefer the use of individual .visit() methods on pages.
+  // example:
+  //   var foo = new FooPage(project);
+  //   foo.visit();
+  topMenu: {
+    clickDocumentation: function() {
+      return clickHelpMenuChild(by.id('help-dropdown'), by.css('.dropdown.open .uib-dropdown-menu'), 0);
+    },
+    clickCLI: function() {
+      return clickHelpMenuChild(by.id('help-dropdown'), by.css('.dropdown.open .uib-dropdown-menu'), 1);
+    },
+    clickAbout: function() {
+      return clickHelpMenuChild(by.id('help-dropdown'), by.css('.dropdown.open .uib-dropdown-menu'), 2);
+    },
+    clickLogout: function() {
+      return clickHelpMenuChild(by.id('iser-dropdown'), by.css('.dropdown.open .uib-dropdown-menu'), 0);
+    }
```

Comment:
Can we find the item by the label? Doing it by index will break if we ever add items in the middle or reorder the menu.